src/lib.rs

```diff
@@ -51,18 +54,25 @@ fn merge_results(result: &mut CovResult, result2: &mut CovResult) {
         };
     }
 
-    for (&(line_no, number), &taken) in &result2.branches {
-        match result.branches.entry((line_no, number)) {
+    for (line_no, taken) in result2.branches {
+        match result.branches.entry(line_no) {
             btree_map::Entry::Occupied(c) => {
-                *c.into_mut() |= taken;
+                let v = c.into_mut();
+                for (x, y) in taken.iter().zip(v.iter_mut()) {
```

Comment:
You're handling the case where taken has more elements than v, but what if the opposite happens?

Reply:
if len(v) >= len(taken) then only the n (=len(taken)) elements of v are modified

Reply:
:+1: